ledger/src/blockstore_processor.rs

```diff
@@ -171,6 +172,18 @@ fn execute_batch(
         ..
     } = tx_results;
 
+    let executed_transactions = execution_results
+        .iter()
+        .zip(batch.sanitized_transactions())
+        .filter_map(|(execution_result, tx)| {
+            if execution_result.was_executed() {
+                Some(tx)
+            } else {
+                None
+            }
+        })
```

Comment:
A more concise version:
```suggestion
        .filter_map(|(execution_result, tx)| execution_result.was_executed().then_some(tx))
```